src/api/libcellml/logger.h

```diff
@@ -38,23 +39,60 @@ class LIBCELLML_EXPORT Logger
     Logger &operator=(Logger rhs) = delete; /**< Assignment operator */
 
     /**
-     * @brief Clear the errors from the logger.
+     * @brief Clear the issues from the logger.
      *
-     * Clear the errors from the logger.
+     * Clear the issues from the logger.
      */
-    void removeAllErrors();
+    void removeAllIssues();
 
     /**
-     * @brief Add an error to the logger.
+     * @brief Add an issue to the logger.
      *
-     * Adds the argument @p error to this logger.
+     * Adds the argument @p issue to this logger.
      *
-     * @param error The @c ErrorPtr to add.
+     * @param issue The @c IssuePtr to add.
      */
-    void addError(const ErrorPtr &error);
+    void addIssue(const IssuePtr &issue);
 
     /**
-     * @brief Get the number of errors.
+     * @brief Get the number of issues.
+     *
+     * Return the number of issues of any level currently stored in the logger.
+     *
+     * @return The number of issues.
+     */
+    size_t issueCount() const;
+
+    /**
+     * @brief Get issue of any level at the specified @p index.
+     *
+     * Returns an issue at the @p index.  If the @p index
+     * is not valid a @c nullptr is returned, the valid range for the @p index
+     * is [0, \#issues).
+     *
+     * @param index The index of the issue to return.
+     *
+     * @return A reference to the issue at the given index on success, @c nullptr otherwise.
+     */
+    IssuePtr issue(size_t index) const;
+
+    /**
+     * @brief Get issue of specified @p level at the specified @p index.
+     *
+     * Returns an issue at the @p index of specified @p level.  If the @p index
+     * is not valid a @c nullptr is returned, the valid range for the @p index
+     * is [0, \#issues_of_level).
+     *
+     * @param index The index of the issue to return.
+     * @param level The level of issue to search.
+     *
+     * @return A reference to the issue from the list of issues at the given level,
+     * at the given index on success, @c nullptr otherwise.
+     */
+    IssuePtr issue(size_t index, libcellml::Issue::Level level) const;
```

Comment:
Isn't this method redundant with the `error()`, `warning()` and `hint()` methods below? I would simply remove it.